src/util/MmapVectorImpl.h

```diff
@@ -43,6 +43,8 @@ void MmapVector<T>::writeMetaDataToEnd() {
   ofs.write((char*)&_bytesize, sizeof(_bytesize));
   ofs.write((char*)&MagicNumber, sizeof(MagicNumber));
   ofs.write((char*)&Version, sizeof(Version));
+  ofs.flush();
```

Comment:
I see this pattern over and over again in the code:
Usage of explicit open() and close() for ad_utility::File and std::fstream.
But actually I think this not necessary when the file class is only used locally and the flush and close are done automagically by the destructor. So wouldn't it be more C++ to let RAII do its job here?

Reply:
Makes sense, I played around with this searching for a bug that broke the MmapVectorTest. Ultimately it was something else. So yeah makes sense to let that happen implicitly